### ISSUE TITLE
rshift32 import issue with Bitwise

### ISSUE THREAD

**Issue by nicwest** [vim-jp/vital.vim#537]
I'm seeing an issue when running my tests on travis where `Bitwise.rshift32` (and presumably `Bitwise.lshift32`) isn't defined when the vital module is imported.

```
./vim-themis/bin/themis --reporter dot -r tests

Error occurred in script loading.
function <SNR>21_import(name, ...) abort dict  (~/build/nicwest/vim-generate/autoload/vital/generate.vim)
function <SNR>21__import(name) abort dict  Line:18  (~/build/nicwest/vim-generate/autoload/vital/generate.vim)
 18:       throw 'vital: fail to call ._vital_loaded(): ' . v:exception
vital: fail to call ._vital_loaded(): Vim(return):E700: Unknown function: s:rshift32
make: *** [test] Error 1
```

https://travis-ci.org/nicwest/vim-generate/jobs/268990406
https://github.com/nicwest/vim-generate/tree/13dbcfe0b3097d29c28080d06f2de195b5c039c4

I fixed it in my repo with this change:

https://github.com/nicwest/vim-generate/commit/75b7214032fea99b5b9f8f5d8bf343858701f9c9

I don't know if there is a better way to go about this, or maybe it's an artifact of the way I'm using vital?

**Comment by tyru** [comment]
`rshift32()` (and `lshift32()`) is added in 194cb7de8e2fd7f03a33e06960fcc0795d19a429 (#444).
But you seem to use older version 714f3966cc17293a44c785399b768ae244cd6915 (https://github.com/nicwest/vim-generate/blob/master/autoload/vital/generate.vital).
Please update vital by `:Vitalize` in your repo.

**Comment by nicwest** [comment]
@tyru I updated and the issue is still present:
https://travis-ci.org/nicwest/vim-generate/builds/269493589
https://github.com/nicwest/vim-generate/commit/8ba65f4361e55864c1bb17ec2d2de2b0a72b10f5

**Comment by tyru** [comment]
@nicwest What is your `:echo has('num64')` output?

Hmm, this seems a bug of vitalizer when `has('num64') == 0` environment. (cc: @thinca, @haya14busa)
Then rshift32() is defined as `a:module.rshift32()` in `s:_vital_created()`.

https://github.com/nicwest/vim-generate/blob/8ba65f4361e55864c1bb17ec2d2de2b0a72b10f5/autoload/vital/_generate/Bitwise.vim#L89

But `vital#_generate#Bitwise#import()` returns `s:rshift32` as a function name.

https://github.com/nicwest/vim-generate/blob/8ba65f4361e55864c1bb17ec2d2de2b0a72b10f5/autoload/vital/_generate/Bitwise.vim#L5

**Comment by nicwest** [comment]
@tyru looks like `has('num64') == 0` on the travis machine:

https://travis-ci.org/nicwest/vim-generate/builds/269532135
https://github.com/nicwest/vim-generate/commit/627a8ac85f095c5d6c96b229b29e9545400cd899

**Comment by thinca** [comment]
This is vital's bug.  Thank you for report.
This will may happen if we define functions in `s:_vital_created()`.
This is related to autoloadize. @haya14busa Could you fix it?

**Comment by haya14busa** [comment]
@nicwest i'm sorry for late response. I fixed this by #553. Can you update vital.vim? Thanks!

**Comment by nicwest** [comment]
@haya14busa will do, thanks!
